vault/seal.go

```diff
@@ -484,6 +485,9 @@ func readStoredKeys(ctx context.Context, storage physical.Backend, encryptor *se
 
 	pt, err := encryptor.Decrypt(ctx, blobInfo, nil)
 	if err != nil {
+		if strings.Contains(err.Error(), "message authentication failed") {
+			return nil, &ErrInvalidKey{Reason: "failed to decrypt keys from storage"}
```

Comment:
What do you think about including the error in the failed reason?

Reply:
To me the 'message authentication failed' part does not add any value. But to other maybe it does? I'm fine either way.
As a reminder, this is how the different options now look. Let me know which one is preferable:

Main branch:

> Error unsealing: Error making API request.
>
> URL: PUT http://127.0.0.1:8200/v1/sys/unseal
> Code: 500. Errors:
>
> * unable to retrieve stored keys: failed to decrypt keys from storage: cipher: message authentication failed

Current patch set:

>Error unsealing: Error making API request.
>
>URL: PUT http://127.0.0.1:8200/v1/sys/unseal
>Code: 400. Errors:
>
>* unable to retrieve stored keys: invalid key: failed to decrypt keys from storage

If we add the error in the reason:
>Error unsealing: Error making API request.
>
>URL: PUT http://127.0.0.1:8200/v1/sys/unseal
>Code: 400. Errors:
>
>* unable to retrieve stored keys: invalid key: failed to decrypt keys from storage: cipher: message authentication failed

Reply:
Thanks for providing example outputs. The reason I asked for including the error is that there could be other code paths that fail. Then two failures would have similar prefix of `unable to retrieve stored keys: invalid key: failed to decrypt keys from storage`. Adding the error would differentiate a bit as to what happened.

Reply:
Yeah, makes sense. I added the error.